src/logic/transaction.ts

```diff
@@ -127,6 +134,67 @@ export async function checkTransactionData(
   if (!(await contracts.isValidContractAddress(contractAddress))) {
     throw new Error(`Invalid contract address "${contractAddress}"`)
   }
+
+  const minPrice = await config.requireNumber('MIN_SALE_VALUE')
```

Comment:
I would move all of this logic to a specific validator file like
`saleValidator.ts` and call it:

```
Sales.validateTx(tx)
```